tool_water_balance/tests/test_waterbalance.py

```diff
@@ -53,17 +125,327 @@ def setUp(self):
             ),
             QgsProject.instance(),
         )
-        self.polygon.transform(tr)
+        polygon.transform(tr)
+    assert polygon.isGeosValid(), "polygon is GeoInvalid. WaterBalance tests will fail"
+    return polygon
+
+
+@pytest.fixture()
+def ts_datasources():
```

Comment:
Er is ook al een algemene `ts_datasources` fixture in de `conftest.py` in de root. Is dit hetzelfde?